init/eessi_archdetect.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# current pathways implemented in EESSI
+# x86_64/generic
+# x86_64/intel/haswell
+# x86_64/intel/skylake_avx512
+# x86_64/amd/zen2
+# x86_64/amd/zen3
+# aarch64/generic
+# ppc64le/generic
+# ppc64le/power9le
+
+ARGUMENT=${1:-none}
+
+cpupath () {
+  # let the kernel tell base machine type
+  MACHINE_TYPE=${EESSI_MACHINE_TYPE:-$(uname -m)}
+
+  # fallback path
+  CPU_PATH="${MACHINE_TYPE}/generic"
+
+  if [ ${MACHINE_TYPE} == "aarch64" ]; then
+    echo ${CPU_PATH}
+    exit
+  fi
+
+  if [ ${MACHINE_TYPE} == "ppc64le" ]; then
+    PROC_CPUINFO=${EESSI_PROC_CPUINFO:-/proc/cpuinfo}
+    CPU_FLAGS=$(grep -m 1 -i ^cpu ${PROC_CPUINFO})
+    [[ $CPU_FLAGS =~ .*POWER9* ]] && HAS_POWER9=true
+
+    [[ ${HAS_POWER9} ]] && CPU_PATH=${MACHINE_TYPE}/power9le
+    echo ${CPU_PATH}
+    exit
+  fi
+
+  if [ ${MACHINE_TYPE} == "x86_64" ]; then
+    # check for vendor info, if available, for x86_64
+    PROC_CPUINFO=${EESSI_PROC_CPUINFO:-/proc/cpuinfo}
+    CPUINFO_VENDOR_FLAG=$(grep -m 1 -i ^vendor ${PROC_CPUINFO})
+    [[ $CPUINFO_VENDOR_FLAG =~ .*GenuineIntel* ]] && CPU_VENDOR=intel
+    [[ $CPUINFO_VENDOR_FLAG =~ .*AuthenticAMD* ]] && CPU_VENDOR=amd
+
+    CPU_FLAGS=$(grep -m 1 -i ^flags ${PROC_CPUINFO})
+    [[ $CPU_FLAGS =~ .*avx2* ]] && HAS_AVX2=true
+    [[ $CPU_FLAGS =~ .*fma* ]] && HAS_FMA=true
```

Comment:
We need to be a bit more careful here, `fma` could be (in theory) a partial match in another CPU feature, so spaces before/after are needed here (while taking into account that `fma` may be listed last, so no trailing space in that case...)